raw/gcs_data_consistency_checker.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import typing as t
+from datetime import datetime, timedelta
+from google.cloud import storage
+
+BUCKET = "gcp-public-data-arco-era5"
+START_YEAR = 1979
+END_YEAR = 2022
+
+
+def list_files_in_directory(bucket_name: str, directory_prefix: str) -> t.List[str]:
+    """Lists files in a specific directory within a Google Cloud Storage bucket."""
+    client = storage.Client()
+    bucket = client.get_bucket(bucket_name)
+    return [blob.name for blob in bucket.list_blobs(prefix=directory_prefix)]
+
+
+def find_extra_files(directory_files: t.List[str], desired_files: t.List[str]
+                     ) -> t.List[str]:
+    """Returns a list of files that are in the directory but not in the desired list."""
+    return [file for file in directory_files if file not in desired_files]
+
+
+def generate_daily_file_names(year: int) -> t.List[str]:
+    """Generates a list of daily file names for a given year."""
+    generated_files = []
+    current_date = datetime(year, 1, 1)
+    end_date = datetime(year + 1, 1, 1)
+    delta = timedelta(days=1)
+
+    while current_date < end_date:
+        formatted_date = current_date.strftime('%Y%m%d')
+        generated_files.append(
+            f'raw/ERA5GRIB/HRES/Daily/{year}/{formatted_date}_hres_dve.grb2')
+        generated_files.append(
+            f'raw/ERA5GRIB/HRES/Daily/{year}/{formatted_date}_hres_o3q.grb2')
+        generated_files.append(
+            f'raw/ERA5GRIB/HRES/Daily/{year}/{formatted_date}_hres_qrqs.grb2')
+        generated_files.append(
+            f'raw/ERA5GRIB/HRES/Daily/{year}/{formatted_date}_hres_tw.grb2')
+        generated_files.append(
+            f'raw/ERA5GRIB/HRES/Daily/{year}/{formatted_date}_hres_tuvw.grb2')
+        current_date += delta
```

Comment:
really clever and pretty clean! I do think that this could be made a bit simpler with `pd.date_range()`.